### ISSUE TITLE
firenvim temp files restored when nvim opened

### ISSUE THREAD

**Issue by Praful** [glacambre/firenvim#1621]
I've checked the troubleshooting guide.

- OS Version: Linux Mint 21.3
- Browser Version: Firefox 127.0.1
- Browser Addon Version: 0.2.16
- Neovim Plugin Version:0.2.15

### What I tried to do
Use the firenvim browser plugin in firefox. This works fine.

### What happened
When I open neovim, there are several buffers that have been created, I assume, by firenvim. Here are some of them, produced by `ls`:

```
2      "/run/user/1000/firenvim/web.whatsapp.com__-2-DIV-1-DIV-2-DIV-1-DIV-1-DIV-1_2024-06-24T23-07-34-537Z.txt" line 1
3      "/run/user/1000/firenvim/chatgpt.com_3641-ce7e-4aa4-9a0c-43b848034d0f_TEXTAREA-id-prompt-textarea_2024-06-24T23-06-54-244Z.txt" line 1
4      "/run/user/1000/firenvim/chatgpt.com_3641-ce7e-4aa4-9a0c-43b848034d0f_TEXTAREA-id-prompt-textarea_2024-06-24T23-05-55-710Z.txt" line 4
5      "/run/user/1000/firenvim/web.whatsapp.com__-2-DIV-1-DIV-2-DIV-1-DIV-1-DIV-1_2024-06-24T23-03-38-773Z.txt" line 1
6      "/run/user/1000/firenvim/web.whatsapp.com__-2-DIV-1-DIV-2-DIV-1-DIV-1-DIV-1_2024-06-24T23-03-06-178Z.txt" line 1
7      "/run/user/1000/firenvim/chatgpt.com_3641-ce7e-4aa4-9a0c-43b848034d0f_TEXTAREA-id-prompt-textarea_2024-06-24T23-02-35-371Z.txt" line 1
8      "/run/user/1000/firenvim/web.whatsapp.com__-2-DIV-1-DIV-2-DIV-1-DIV-1-DIV-1_2024-06-24T23-00-04-737Z.txt" line 1
9      "/run/user/1000/firenvim/web.whatsapp.com__-2-DIV-1-DIV-2-DIV-1-DIV-1-DIV-1_2024-06-24T22-58-50-781Z.txt" line 1
```
Is there a way to stop these buffers being persisted or opened in neovim? They also open in neovide.

**Comment by glacambre** [comment]
If the problem also happens with Neovide, the problem is in your config, not in Firenvim. Try to remove your plugins one by one until the problem stops happening. Please do report here if you figure out that it's really a Firenvim problem, but I doubt you will.

**Comment by glacambre** [comment]
>The issue was the set viminfo in my vimrc

Huh, interesting, thanks for sharing. I didn't even know Neovim could do this!

>On another note, the second autosave function on the home page didn't work

Thanks a lot for reporting this. I probably broke it when I converted the example from vimscript to lua.  I fixed it in 992e57666c0d4c301c5a231354b8237d7bae0282 and took the liberty of marking you as the author of the fix, I hope that's okay :).

**Comment by Praful** [comment]
Thanks @glacambre :)
For the function, you may want to include (or provide as comment):
```
if not vim.g.started_by_firenvim or vim.g.timer_started then
    return
end
```
Otherwise the timed write will happen even when the user is in nvim/neovide as well as 
firenvim!
